docs/reference/mpyfiles.rst

```diff
@@ -0,0 +1,110 @@
+.. _mpy_files:
+
+MicroPython .mpy files
+======================
+
+MicroPython defines the concept of an .mpy file which is a binary container
+file format that holds precompiled code, and which can be imported like a
+normal .py module.  The file ``foo.mpy`` can be imported via ``import foo``,
+as long as ``foo.mpy`` can be found in the usual way by the import machinery
+(usually if it is found in a directory listed in ``sys.path``).
+
+These .mpy files can contain bytecode which is usually generated from Python
+source files (.py files) via the ``mpy-cross`` program.  For some architectures
+an .mpy file can also contain native machine code, which can be generated in
+a variety of ways, most notably from C source code.
+
+A given .mpy file may or may not be compatible with a given MicroPython system.
+Compatibility is based on the following:
```

Comment:
If someone was having trouble with an incompatible .mpy file and they found this section, so perhaps some hints on how to deal with (likely) an incorrect mpy-cross version or invocation.

* If using an official MicroPython release version, then checkout the repo at that tag before building mpy-cross.

* Look at the Makefile for your MicroPython port and check the `MPY_CROSS_FLAGS` variable to find the corresponding platform-specific values of flags such as `-march` and `-mcache-lookup-bc`.